chore: cleanup

anuragxxd opened the pull request:
## Description
<!-- Please include a summary of the change and the relevant issue(s) it resolves, if any (otherwise delete that line). -->

Fixes #147 #139 #269 #268 #244 #243 #238

## Checklist
<!-- Please go through the following checklist to ensure that your change is ready for review. -->

- [ ] My code follows the [contributing guidelines][contributing] of this project.
- [ ] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments
<!-- If there are unchecked boxes in the list above, but you would still like your PR to be reviewed or considered for merging, please describe here why boxes were not checked. -->

[contributing]: CONTRIBUTING.md

git-anurag-hub commented:
Hey @SalihuDickson @JaeAeich, I'd appreciate it if you could take a look at this PR. I've cleaned things up by replacing older packages with their newer counterparts. Since this is a large PR, here are the key areas to focus on:
- @SalihuDickson I removed the deprecated dependency from the build script that was causing errors. I also updated the make-react script's component path to address another error. To verify these changes, it would be helpful if you could run the script locally.
- @JaeAeich I've made some changes to WES & TES as well. If you have time, it would be great if you could run them locally to ensure they're working as expected.
- @JaeAeich We're removing the docs entirely in favor of #262  so there's no need to migrate them. Could you consider updating your PR to use the latest packages?
- commit-lint & commit-generator: To streamline our squash & merge strategy, I've removed these from the recommit.

I'm also considering moving TRS & TRS-filer to a legacy folder to deprecate them before migrating them to the lit & design package implementation. What are your thoughts on this approach?

For peace of mind, it would be great if you both could try running the PR locally to check any functionalities it might affect. While I've tested everything myself, more reviews are always valuable. thanks!

JaeAeich commented on packages/ecc-client-elixir-trs-filer/playwright.config.js:
Lets not touch filer and trs

JaeAeich commented on package.json:
```diff
@@ -17,19 +17,16 @@
     "test": "turbo run test --parallel",
     "test:watch": "turbo run test --watch",
-    "clean": "rm -rf node_modules && rm -rf */*/node_modules",
+    "clean": "rm -rf node_modules && turbo run clean",
```
This mean all package.json need to have a `clean` command (would need to add in handlebars files). Also if we remove node_modules we wont have turbo as dependecy :) so the command will fail.

anuragxxd replied:
Ahh! Good catch I had turbo installed globally maybe that's the reason everything worked. Have change the command to:
```
"clean": "turbo run clean && rm -rf node_modules",
```

JaeAeich commented on packages/ecc-client-ga4gh-tes/.eslintignore:
```diff
@@ -1,22 +1,32 @@
 README.md
 CONTRIBUTORS.md
 CODE_OF_CONDUCT.md
+
```
auto generated ig.

JaeAeich commented on packages/ecc-client-ga4gh-tes/README.md:
For individual readmes of files lets just have description and nothing more, people should refer from documentation. This will be hard to maintain otherwise.

JaeAeich commented on packages/ecc-client-ga4gh-tes/package.json:
```diff
@@ -1,63 +1,57 @@
+    "@lit/react": "*",
+    "react": "*",
+    "commander": "*",
+    "custom-element-jet-brains-integration": "*",
```
if we are doing clean up lets figure out if these are needed? @SalihuDickson

SalihuDickson replied:
yh they are needed, the react packages are needed, `@lit/react` and `react` are needed for packaging the react components. commander is needed to run the cem config that generates the manifest so that vscode, jetbrains, and some other supported  editors know how to handle the components. And the last package is also needed for generating the jetbrains manifest for custom components. I think there's even one package missing. I'll have to look into it.

JaeAeich replied:
But the point is do we need manifest files in the first place, if I am not wrong it can be used in very specific part of debugging, I have rarely used that kind of debugging.

SalihuDickson replied:
well we don't **need** the manifest but it's a nice to have. For now the main plus is that it allows devs working with our components to enable autocomplete in regular html and other supported frameworks, and hover over components to get more info, as well as a documentation link to jump to the docs for a specific component, the same way you can jump to an mdn reference for regular components. You can read more about it [here](https://github.com/microsoft/vscode-custom-data) and [here](https://github.com/JetBrains/web-types). 

It also allows the possibility of developing tools that can take advantage of a properly defined web compnents manifest standard. For example we're currently using the regular manifest to  auto generate react components.

anuragxxd replied:
Agree with @SalihuDickson here, it's best to have them generated for the devs using our library.

JaeAeich commented on packages/ecc-client-ga4gh-tes/package.json:
```diff
@@ -1,63 +1,57 @@
-    "eslint-plugin-prettier": "^4.2.1",
-    "rimraf": "^3.0.2",
+    "eslint-config-elixir": "*",
+    "eslint-config-prettier": "^8.3.0",
```
Can we have all the common deps as global deps, ie *

anuragxxd replied:
Yeah, that must be the ideal case! I just don't think it is the best idea to do this in the same PR as this, as that may require more discussion & maybe some changes to the source.

JaeAeich commented on packages/ecc-client-ga4gh-tes/package.json:
```diff
@@ -1,63 +1,57 @@
-    "dev": "npm run build && concurrently -k -r \"npm run build -- --watch\" \"wds\"",
-    "lint:fix": "npm run lint -- --fix",
-    "lint": "eslint --ext .ts,.html . --ignore-path .gitignore",
+    "analyze": "cem analyze --litelement",
```
do we need this? why is it here, lets remove if we aren't going to.

JaeAeich commented on packages/ecc-client-ga4gh-tes/package.json:
```diff
@@ -1,63 +1,57 @@
-    "lint:fix": "npm run lint -- --fix",
-    "lint": "eslint --ext .ts,.html . --ignore-path .gitignore",
+    "analyze": "cem analyze --litelement",
+    "build": "node ../../scripts/build.js",
```
how about adding clean command just before this, IK build script is handeling this but clearly not efficiently.
PS: accordingly lets update the template files as well (handlebars).

anuragxxd replied:
Hey @SalihuDickson, if possible can you maybe make PR from your branch to this branch? It will ease the merging & testing all together!

JaeAeich commented on scripts/make-react.js:
```diff
@@ -5,9 +5,15 @@ const fs = require("fs");
 const { program } = require("commander");
 const path = require("path");
 const prettier = require("prettier");
-const pascalCase = require("pascal-case");
```
please remove pascal case from build script as well, its giving errros and warnings.

SalihuDickson replied:
can you specify, what errors and warnings? If its not tenable maybe I can write a simple algorithm that fits our specific use case.

JaeAeich replied:
my bad!

JaeAeich reviewed (changes requested):
@git-anurag-hub 
- Please change the license from MIT to Apache 2.0 for individual packages as well ([ref](https://github.com/elixir-cloud-aai/cloud-components/issues/269)).
- Please configure the dev server ports to be different they are running on 8000 currently.
- Building the package is really unstable and breaks quite often, maybe the clean up before build should fix it.
- `dev` fails if the user hasn't run `build`, maybe we need to create a custom script that build the dependency of the packages (we can provide our dependecy list or just infer all the deps that have "*"), if they are already built runs the dev else builds the deps and then runs. 
- `husky` if we are keeping that, then lets add [lint-staged](https://www.npmjs.com/package/lint-staged), I mean otherwise I don't see a point of running the test each time I have to commit, yeah would have sense to lint some files (staged).

JaeAeich commented:
> I'm also considering moving TRS & TRS-filer to a legacy folder to deprecate them before migrating them to the lit & design package implementation. What are your thoughts on this approach?

Let's not move them to a different package for the sake of consistency. Instead, we could remove and/or comment out their scripts so that they don't build or run their tests in CI.

JaeAeich commented:
Also please run `npm i` and update the lock file. Apart from that, take a look into this #252 either take reference or configure the package acc you in this PR itself. 

PS: If so please also make the changes in template files.

SalihuDickson commented:
@JaeAeich, I'm not sure how much time @git-anurag-hub has on his hands, and this PR looks like it still needs quite a bit of work and testing before it is ready to me merged. To avoid a lot of back and forth, especially considering the size of the PR, I think we should just make PRs to this fixing the issues we notice.

anuragxxd commented:
> @git-anurag-hub
> 
> * Please change the license from MIT to Apache 2.0 for individual packages as well ([ref](https://github.com/elixir-cloud-aai/cloud-components/issues/269)).
> * Please configure the dev server ports to be different they are running on 8000 currently.
> * Building the package is really unstable and breaks quite often, maybe the clean up before build should fix it.
> * `dev` fails if the user hasn't run `build`, maybe we need to create a custom script that build the dependency of the packages (we can provide our dependecy list or just infer all the deps that have "*"), if they are already built runs the dev else builds the deps and then runs.
> * `husky` if we are keeping that, then lets add [lint-staged](https://www.npmjs.com/package/lint-staged), I mean otherwise I don't see a point of running the test each time I have to commit, yeah would have sense to lint some files (staged).
> * `prettier` please can we also configure prettier in this PR itself, only formatting md, mdx, html, css, JSON and other non code files only though. And since this can be added to husky, [this](https://prettier.io/docs/en/precommit.html) might be useful.

@JaeAeich 
- Sounds good!  I've updated the LICENSE files in all individual packages to Apache 2.0.
- Seems like all are pointed to the different ports, but still have reconfigured and arranged them more sensibly.
- I guess (atleast I hope) this will not be the problem after the refactor is completed. In the meantime, we'll continue investigating and resolving any build-related problems.
- I think the better way of this will be for the dev's to start the dev env for the dependencies (not the build) so that they they can leverage features like HMR for better debugging. Meanwhile will explore automating this process by configuring `turbo.json`, similar to the build command.
- Good idea, this is the reason why we have removed most of the pre-commit hooks now to reduce the overhead from the developers but I guess it is not the best idea to use the Github runner minutes for the test & lint scripts. 
- We can have issue for this!

anuragxxd commented:
> @JaeAeich, I'm not sure how much time @git-anurag-hub has on his hands, and this PR looks like it still needs quite a bit of work and testing before it is ready to be merged. To avoid a lot of back and forth, especially considering the size of the PR, I think we should just make PRs to this fixing the issues we notice.
> 
> @git-anurag-hub, please correct me if you would prefer we just request changes. I'm just trying to get this out as soon as possible and it seems it would be easier and quicker for you to look through and approve our PRs instead of having to go and make changes and request reviews to start the process all over again.

Sure @SalihuDickson, while I have addressed most of the issues referenced here. Are there any other issues you think which are left here? If so please let me know. I have tested most of the functionality affected by this PR but want you & @JaeAeich  to take a look at the parts where you have worked. Saying this please go ahead & fix the issues that you might think can take less time solving parallely.

JaeAeich commented:
@anuragxxd for some reason, I can't build the packages propoerly, maybe we need to have discussion on this. Is it working fine on your end. I did get it running but that seems to be an on off thing, its not consistent. I can't be sure the packages will definetely build everytime.

JaeAeich commented:
> @anuragxxd for some reason, I can't build the packages propoerly, maybe we need to have discussion on this. Is it working fine on your end. I did get it running but that seems to be an on off thing, its not consistent. I can't be sure the packages will definetely build everytime.

Please merge the fix and some more cleanup from #285.

JaeAeich reviewed (approved):
LGTM :rocket:

Small PRs from now on please :face_exhaling: :rofl: !

JaeAeich commented on turbo/generators/template-package/package.json.hbs:
```diff
@@ -10,6 +10,8 @@
   "scripts": {
     "build": "",
+    "clean": "rm -rf node_modules",
+    "clean:react": "",
```
no longer needed right!

JaeAeich replied:
done

JaeAeich commented on packages/ecc-utils-design/package.json:
```diff
@@ -17,23 +16,23 @@
+    "url": "https://github.com/elixir-cloud-aai/cloud-components.git"
+  },
+  "customElements": "dist/custom-elements.json",
+  "files": ["dist", "README.md", "package.json", "LICENSE"],
```
maybe a nit pick but since we already have such huge changes, why not format these small things as well. Maybe adding a linter would have been a good idea, but as you guy think that it can a separate issue then ig so.

But regardless can we still format it with our IDE's and will add the formatter later.
PS: prettier is a dep already in root dir.

JaeAeich commented:
LGTM, Everything seems to work on my machine, please resolve the above as you see fit and lets merge :rocket:

JaeAeich commented on packages/ecc-utils-design/package.json:
```diff
@@ -17,23 +16,23 @@
   "scripts": {
     "analyze": "cem analyze --litelement",
-    "build": "node ../../scripts/build.js",
+    "build": "node ../../scripts/build.js -p ecc-utils-design-",
```
From the script it seems we still need it!

JaeAeich commented on scripts/build.js:
```diff
@@ -42,22 +48,27 @@ nextTask("Cleaning up previous build", () => {
+// rewrite to ask for permission for each dependency that isnr present or just throw error
+// add option for user to add prettier config
+// reading from package json for some reason does not block other processes
 nextTask("installing dependencies", async () => {
```
please take a look at this task, this is rewriting package.json which firstly I don;t think is supposed to be done, secondly if the whole purpose it to add some deps, why not add it to the handlebars and be done with it. Then we will only have to install dependencies here. This task is the one that is causing change in format of package.json. Am I missing something? @SalihuDickson @anuragxxd

SalihuDickson replied:
Sure, I guess that would be a better approach. This was always meant to be a temporary approach and was only viable since it was an internal tool. But you're right, we shouldn't be rewriting package json, I'll fix this.

SalihuDickson replied:
@JaeAeich, #290 fixes this, please review.

JaeAeich commented on packages/ecc-client-ga4gh-tes/package.json:
```diff
@@ -1,63 +1,57 @@
-  "description": "Web components for TES",
+  "description": "Web Component for interactive with TES API",
+  "license": "Apache-2.0",
+  "version": "0.0.0",
```
Also "interacting", not "interactive" in desc

anuragxxd commented:
@SalihuDickson @JaeAeich Alright, gonna go ahead and merge this if it all looks good to you! I tested the changes locally and everything seems to be working fine on my end.

anuragxxd commented:
@SalihuDickson @JaeAeich alright, gonna go ahead and merge this if it all looks good to you. have tested the changes on my machine too seems to work fine.

SalihuDickson reviewed (approved):
lgtm
